src/commands/registries/azure/deployImageToAzure.ts

```diff
@@ -141,3 +148,68 @@ class DockerSiteCreateStep extends AzureWizardExecuteStep<IAppServiceWizardConte
         return !context.site;
     }
 }
+
+class DockerWebhookCreateStep extends AzureWizardExecuteStep<IAppServiceWizardContext> {
+    public priority: number = 141; // execute after DockerSiteCreate
+
+    private _treeItem: RemoteTagTreeItemBase;
+
+    public constructor(treeItem: RemoteTagTreeItemBase) {
+        super();
+        this._treeItem = treeItem;
+    }
+
+    public async execute(context: IAppServiceWizardContext, progress: Progress<{ message?: string; increment?: number }>): Promise<void> {
+        const creatingNewApp: string = `Creating webhook for webapp "${context.newSiteName}"...`;
+        ext.outputChannel.appendLine(creatingNewApp);
+        progress.report({ message: creatingNewApp });
+
+        const site: Site = nonNullProp(context, 'site');
+        let siteClient = new SiteClient(site, context);
+        let appUri: string = (await siteClient.getWebAppPublishCredential()).scmUri;
+        if (this._treeItem instanceof AzureTagTreeItem) {
+            await this.createWebhookForApp(this._treeItem, context, appUri);
+        } else if (this._treeItem instanceof DockerHubTagTreeItem) {
+            // point to dockerhub to create a webhook
+            // http://cloud.docker.com/repository/docker/<registryName>/<repoName>/webHooks
+            const dockerhubPrompt: string = "Copy web app endpoint and browse to dockerhub";
+            let response: string = await vscode.window.showInformationMessage("Please browse to your dockerhub account to set up a CI/CD webhook", dockerhubPrompt);
+            if (response) {
+                await vscode.env.clipboard.writeText(appUri);
+                await openExternal(`https://cloud.docker.com/repository/docker/${this._treeItem.parent.parent.parent.username}/${this._treeItem.parent.repoName}/webHooks`);
+            }
+        }
+
+    }
+
+    public shouldExecute(context: IAppServiceWizardContext): boolean {
+        return !!context.site;
+    }
+
+    private async createWebhookForApp(node: AzureTagTreeItem, wizardContext: IActionContext & Partial<IAppServiceWizardContext>, appUri: string): Promise<void> {
+        // fields derived from the app service wizard
+        let siteName: string = wizardContext.site.name;
+        let webhookName: string = `webapp${siteName}`;
+
+        // variables derived from the container registry
+        const crmClient = createAzureClient(node.parent.parent.parent.root, ContainerRegistryManagementClient);
+        let registryResourceGroupName: string = node.parent.parent.resourceGroup;
+        let registryName = node.parent.parent.registryName;
+
+        const existingWebhooks = await crmClient.webhooks.list(registryResourceGroupName, registryName);
+        if (existingWebhooks.find((hook) => hook.name === webhookName)) {
+            return;
```

Comment:
The way it is now, the UI will say "Creating webhook..." but never alert the user that the webhook already exists.  I think it'd be confusing for users since they would expect the UI to indicate that the webhook creation was resolved whether by already existing or being created.